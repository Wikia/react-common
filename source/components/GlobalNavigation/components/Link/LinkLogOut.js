import React from 'react';
import { useTranslation } from 'react-i18next';
import PropTypes from 'prop-types';

import withRedirectUrl from '../../hocs/withRedirectUrl';

const LinkLogOut = ({ link, redirectUrl, className }) => {
    const [t] = useTranslation();

    return (
        <div id={link.title.key} className={className}>
            <form method="POST" action={link.href}>
                <input
                    type="hidden"
                    name={link['param-name']}
                    value={redirectUrl}
                />
                {/* wds-sign-out__button will make this look like a link */}
                <button
                    type="submit"
                    className="wds-sign-out__button"
                    data-tracking-label={link['tracking-label']}
                >
                    {t(link.title.key)}
                </button>
            </form>
        </div>
    );
};

LinkLogOut.propTypes = {
    className: PropTypes.string,
    link: PropTypes.object.isRequired,
    redirectUrl: PropTypes.string.isRequired,
};

LinkLogOut.defaultProps = {
    className: '',
};

export default withRedirectUrl(LinkLogOut);
