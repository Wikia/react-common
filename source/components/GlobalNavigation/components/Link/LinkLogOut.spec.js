import React from 'react'
import { shallow } from 'enzyme';
import merge from 'lodash/merge';

import LinkLogOut from './LinkLogOut';

const defaultProps = {
    link: {
        type: 'link-logout',
        href: 'https:\/\/www.wikia.com\/logout',
        title: {
            type: 'translatable-text',
            key: 'global-navigation-user-sign-out',
        },
        'param-name': 'returnto',
        'tracking-label': 'account.sign-out',
    },
};

function renderComponent(props) {
    const computedProps = merge({}, defaultProps, props);

    return shallow(<LinkLogOut {...computedProps} />);
}

test('LinkLogOut renders correctly with default params', () => {
    expect(renderComponent().dive()).toMatchSnapshot();
});

test('LinkLogOut renders correctly with additional className', () => {
    expect(renderComponent({ className: 'so-classy' }).dive()).toMatchSnapshot();
});
