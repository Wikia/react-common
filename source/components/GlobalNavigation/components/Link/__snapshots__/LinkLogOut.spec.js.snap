// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkLogOut renders correctly with additional className 1`] = `
<div
  className="so-classy"
  id="global-navigation-user-sign-out"
>
  <form
    action="https://www.wikia.com/logout"
    method="POST"
  >
    <input
      name="returnto"
      type="hidden"
      value="http://localhost/"
    />
    <button
      className="wds-sign-out__button"
      data-tracking-label="account.sign-out"
      type="submit"
    >
      global-navigation-user-sign-out
    </button>
  </form>
</div>
`;

exports[`LinkLogOut renders correctly with default params 1`] = `
<div
  className=""
  id="global-navigation-user-sign-out"
>
  <form
    action="https://www.wikia.com/logout"
    method="POST"
  >
    <input
      name="returnto"
      type="hidden"
      value="http://localhost/"
    />
    <button
      className="wds-sign-out__button"
      data-tracking-label="account.sign-out"
      type="submit"
    >
      global-navigation-user-sign-out
    </button>
  </form>
</div>
`;
